Create Sponsorship App #10 (Add Sponsorship app with profile/tier models, forms, templates, and admin registration)

lidokogi opened the pull request:
Fix #10 

This PR implements the Sponsorship app for the PyLadiesCon portal. It includes the following features:

1) SponsorshipProfile model with:

- One-to-one relationship with User

- Main contact (required)

- Additional contacts (optional)

- Sponsor organization name

- Sponsorship type (Individual or Organization)

- Foreign key to SponsorshipTier

- Logo upload field

- Company description

- Application status (Pending, Approved, Rejected, Cancelled)

2) SponsorshipTier model with:

- Amount

- Name

- Description

3) Admin registration for both models with list displays and filters

4) Forms and views for creating sponsorship profiles

5) Templates: create.html, success.html

6) URL configuration for the app

![Sponsorship Profile Form](https://github.com/user-attachments/assets/f865f8a3-0dbc-41cb-9091-d6d9bedc9b42)
![Django Admin Panel](https://github.com/user-attachments/assets/c286cfe0-6562-47e9-9c2a-9903ea6e8bd6)
![Success Page](https://github.com/user-attachments/assets/74b8ee86-5282-40ba-855a-62398b3ae980)

Mr-Sunglasses reviewed (changes requested):
- Please remove `db.sqlite3` file as is a db file and we don't need it.

```
# Byte-compiled / optimized / DLL files
__pycache__/
*.py[cod]
```

- Please add these above lines in your `.gitignore` file so that they don't track the changer for __pycache__

lidokogi commented:
> * Please remove `db.sqlite3` file as is a db file and we don't need it.
> 
> ```
> # Byte-compiled / optimized / DLL files
> __pycache__/
> *.py[cod]
> ```
> 
> * Please add these above lines in your `.gitignore` file so that they don't track the changer for **pycache**

Done! Thanks for the suggestions.

Mr-Sunglasses commented on portal/settings.py:
```diff
@@ -242,3 +245,5 @@
     # Otherwise, send emails to the console
     EMAIL_BACKEND = "django.core.mail.backends.console.EmailBackend"
+
+DEBUG = True 
```
```suggestion

```

No need to explicitly define DEBUG as it is already defined above in the file.


you just need to export it as an environment variable to use it.

Mr-Sunglasses commented:
> > * Please remove `db.sqlite3` file as is a db file and we don't need it.
> > 
> > ```
> > # Byte-compiled / optimized / DLL files
> > __pycache__/
> > *.py[cod]
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * Please add these above lines in your `.gitignore` file so that they don't track the changer for **pycache**
> 
> Done! Thanks for the suggestions.

To remove __pycache__ file now after commit please reference to this stackoverflow link: https://stackoverflow.com/a/71930352/16563091

Adding it in `.gitignore` helps in preventing adding them in the future commits, for now you have to remove it manually.

lidokogi commented:
> > > * Please remove `db.sqlite3` file as is a db file and we don't need it.
> > > 
> > > ```
> > > # Byte-compiled / optimized / DLL files
> > > __pycache__/
> > > *.py[cod]
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 
> > > * Please add these above lines in your `.gitignore` file so that they don't track the changer for **pycache**
> > 
> > 
> > Done! Thanks for the suggestions.
> 
> To remove **pycache** file now after commit please reference to this stackoverflow link: https://stackoverflow.com/a/71930352/16563091
> 
> Adding it in `.gitignore` helps in preventing adding them in the future commits, for now you have to remove it manually.

Oh I see! I have done that now. Looks good?

Mr-Sunglasses commented on sponsorship/templates/sponsorship/profile_form.html:
```diff
@@ -0,0 +1,29 @@
+{% extends "portal/base.html" %}
```
Please move this `sponsorship/templates` to `templates` folder in the parent directory as we manage all the templates for the project there. you can name the folder `sponsorship` in the parent `templates` directory.

lidokogi replied:
Thanks. Now taken care of!

Mr-Sunglasses commented on portal/settings.py:
```diff
@@ -28,9 +31,7 @@
-ALLOWED_HOSTS = os.environ.get("DJANGO_ALLOWED_HOSTS")
-if ALLOWED_HOSTS:
-    ALLOWED_HOSTS = ALLOWED_HOSTS.split(",")
+ALLOWED_HOSTS = ['localhost', '127.0.0.1']
```
```suggestion
ALLOWED_HOSTS = os.environ.get("DJANGO_ALLOWED_HOSTS")
if ALLOWED_HOSTS:
    ALLOWED_HOSTS = ALLOWED_HOSTS.split(",")
```

Please revert it back and  export it as an environment variable.

Mr-Sunglasses commented on portal/settings.py:
```diff
@@ -9,6 +9,9 @@
 For the full list of settings and their values, see
 https://docs.djangoproject.com/en/5.1/ref/settings/
 """
+from dotenv import load_dotenv
```
```suggestion

```
No need for it as we are reverting back.

Mr-Sunglasses commented on portal/settings.py:
```diff
@@ -9,6 +9,9 @@
 https://docs.djangoproject.com/en/5.1/ref/settings/
 """
+from dotenv import load_dotenv
+load_dotenv()
```
```suggestion

```

Mr-Sunglasses commented on portal/settings.py:
```diff
@@ -9,6 +9,9 @@
 """
+from dotenv import load_dotenv
+load_dotenv()
+
```
```suggestion
```

Mariatta commented on volunteer/migrations/0002_remove_volunteerprofile_coc_agreement_and_more.py:
Why were the migrations file removed?

lidokogi replied:
0002_remove_volunteerprofile_coc_agreement_and_more.py migration file had issues due to broken references and a malformed field. I removed the old migrations to fix the migration history, then re-ran makemigrations and migrate to regenerate a clean, working migration. Everything should now be resolved and stable again!

Mariatta commented on sponsorship/models.py:
```diff
@@ -0,0 +1,45 @@
+        (ORGANIZATION, 'Organization/Company'),
+    ]
+
+    APPLICATION_PENDING = 'pending'
```
Perhaps the application status can be an enum, or a StrEnum

https://docs.python.org/3/library/enum.html#enum.StrEnum

lidokogi replied:
Sounds good. Just commited that.

Mariatta commented on sponsorship/models.py:
```diff
@@ -0,0 +1,45 @@
+    APPLICATION_REJECTED = 'rejected'
+    APPLICATION_CANCELLED = 'cancelled'
+
+    APPLICATION_STATUS_CHOICES = [
```
This is great!

lidokogi replied:
Thankyou :)

Mariatta commented on sponsorship/models.py:
```diff
@@ -0,0 +1,45 @@
+    sponsor_organization_name = models.CharField(max_length=255)
+    sponsorship_type = models.CharField(max_length=20, choices=SPONSORSHIP_TYPE_CHOICES)
+    sponsorship_tier = models.ForeignKey(SponsorshipTier, on_delete=models.SET_NULL, null=True)
+    logo = models.ImageField(upload_to='sponsor_logos/')
```
You might need to add the Pillow dependency in order to be able to use ImageField. eg adding `Pillow==11.1.0` to the requirements.txt file.

lidokogi replied:
Done!

Mariatta commented on sponsor_logos/profile_pic.png:
I think these files should be excluded from the commit.

lidokogi replied:
Done!

Mariatta commented on sponsorship/forms.py:
```diff
@@ -0,0 +1,20 @@
+    class Meta:
+        model = SponsorshipProfile
+        fields = [
+            'main_contact',
```
In the profile creation form, the main_contact should default to the current user, so it should be a readonly field showing the current user's info.

lidokogi replied:
Makes sense! Just edited that.

Mariatta commented on sponsorship/forms.py:
```diff
@@ -0,0 +1,20 @@
+        model = SponsorshipProfile
+        fields = [
+            'main_contact',
+            'additional_contacts',
```
I think this can be excluded for the first pass. We may need to design a way to add additional contacts. For example, the current user may need to invite the additional contacts to register first before they can be added here. (eg this shouldn't be just the dropdown of existing users). So to simplfy the PR, perhaps exclude this.

Mariatta reviewed (commented):
Thank you! I think this is a great start in the right direction. I like it!

lidokogi commented:
Hi @Mariatta,
I've addressed all your review comments—thank you again for the thoughtful feedback! I'd really appreciate it if you could take another look and let me know if there’s anything else you'd like me to revise. Looking forward to getting this PR ready for merge! 😊
